test/e2e/storage/testsuites/provisioning.go

```diff
@@ -283,18 +259,22 @@ func testProvisioning(input *provisioningTestInput) {
 						},
 					}
 				}
-				RunInPodWithVolume(input.cs, claim.Namespace, claim.Name, podName, command, node)
+				RunInPodWithVolume(cs, claim.Namespace, claim.Name, podName, command, node)
 			}
 			go run(firstPodName, "touch /mnt/test/first && while ! [ -f /mnt/test/second ]; do sleep 1; done")
 			go run(secondPodName, "touch /mnt/test/second && while ! [ -f /mnt/test/first ]; do sleep 1; done")
 			wg.Wait()
 		}
-		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
+		testCase.TestDynamicProvisioning()
 	})
 }
 
-// TestDynamicProvisioning tests dynamic provisioning with specified StorageClassTest and storageClass
-func TestDynamicProvisioning(t StorageClassTest, client clientset.Interface, claim *v1.PersistentVolumeClaim, class *storage.StorageClass) *v1.PersistentVolume {
+// TestDynamicProvisioning tests dynamic provisioning with specified StorageClassTest
+func (t StorageClassTest) TestDynamicProvisioning() *v1.PersistentVolume {
+	client := t.Client
+	claim := t.Claim
```

Comment:
could fail w/ nice error if this is nil

Reply:
Not sure if Github is showing me the right context. What can be nil here? `t.Client` and/or `t.Claim`?

Are you worried that they might be accidentally passed as nil because they are now struct members instead of separate function parameters? I agree, this isn't caught explicitly. If this is the only requested change, then I'd prefer to add assertions in a separate PR.

Reply:
I meant t.Claim, though client could be nil as well. either way, the nil error will be caught a few lines below, so yeah these assertions can be added in a separate PR as "nice-to-haves" :+1:

Reply:
As I was rebasing anyway, I added the asserts.